improvement and fixes

AlwinEsch opened the pull request:
To make add-on better usable and implement missing settings features.

AlwinEsch committed:
update settings.xml to enable GUI entries related to region

AlwinEsch committed:
fix within dabdsp wrong coarseCorrector by take back original defines

NOTE: Original defines renamed from Hz, kHz & MHz to have "Value" on end, this to fix conflicts within Windows Visual Studio defined ones.

AlwinEsch commented:
Note about commit "[move libfec parts to own folder](https://github.com/kodi-pvr/pvr.rtlradio/pull/11/commits/0190b7ff6a471f5b901fbe1c273007c4a40dddda)", after this comes much times crash on playbacks.
As in code no differences to before and only on other binary place in add-on lib, it seems thats somewhere a mem write overflow where destroy some mem data in processes.

ksooo commented on pvr.rtlradio/resources/language/resource.language.en_gb/strings.po:
```diff
@@ -126,7 +132,10 @@ msgstr ""
-# 30204 can be reused
+
+msgctxt "#30204"
+msgid "Enable coarse corrector (for receivers with >1kHz error)."
```
Could you add a hint for end users when to set this to which of the three values? What is the ">1kHz error"?

ksooo commented on src/addon.cpp:
```diff
@@ -1627,6 +1633,34 @@ ADDON_STATUS addon::SetSetting(std::string const& settingName,
+    }
+  }
+
+  // dabradio_coarse_corrector
```
This should read `// dabradio_coarse_corrector_type` (copy/paste error I guess)

ksooo commented on src/pvrtypes.h:
```diff
@@ -159,6 +159,16 @@ struct settings
   // Specifies the output gain for the DAB DSP
   float dabradio_output_gain;
 
+  // dabradio_output_gain
```
This comment seems not to fit here (copy/paste from above?)

AlwinEsch committed:
add DAB setting to enable coarse corrector algorithm

AlwinEsch committed:
cleanup reduced libfec to have fac.c & fec.h only

Further cleaned up the code itself with:
- Remove "goto" usage
- Don't set code values by defines
- Removed not needed encode source
- No more malloc/free in process, done on init/deinit now

AlwinEsch commented:
Thanks @ksooo for review, commented places are fixed.

About the settings I increased the text and now shown as this, what you think, OK as this or another idea about?:
![Screenshot 2024-04-01 025326](https://github.com/kodi-pvr/pvr.rtlradio/assets/6879739/f159f34e-7845-4523-8868-6ae7df885ffc)

ksooo commented:
Yes, this is better. Thanks.

AlwinEsch committed:
remove src/compat and move the only used file complex.cpp to src/utils

All other files in src/compat was not used.

AlwinEsch committed:
remove anglepch.cpp and anglepch.h as no more usable

Within Kodi's build system they can not be used in all cases.

AlwinEsch committed:
rename processing code folders start with dsp_

Before was the "dsp" on end of name, this now changed to have folders in better alphabetical flow and no mix with other folders.

AlwinEsch committed:
move / rename "src/stdafx.h" to "src/utils/value_size_defines.h"

As now only macros about value sizes inside becomes it renamed and moved to the utils folder.

AlwinEsch committed:
move gui related parts in own source folder "src/gui"

Also to have more clean and files better sorted.

AlwinEsch committed:
move two another headers to src/utils

AlwinEsch committed:
move all exception handlers to one folder

As there some of them, looks it cleaner to have in own folder about.

AlwinEsch committed:
fix Linux compile errors
